nservicebus/upgrades/5to6-moving-away-from-ibus.md

```diff
@@ -0,0 +1,85 @@
+
+snippet:5to6-endpoint-send-messages-outside-handlers
+
+If the endpoint is hosted using NServiceBus.Host, use the [IWantToRunWhenEndpointStartsAndStops interface](/nservicebus/upgrades/host-6to7.md) as outlined in the NServiceBus.Host documentation.
+
+## Sending messages inside message handlers
+
+Instances of `IBus` that were being injected into message handler classes by the IoC container can be safely deleted. 
+
+The message handler signature now includes an additional `IMessageHandlerContext` parameter, which provides the methods that used to be called from `IBus`. Use the `IMessageHandlerContext` to send and publish messages from within the message handler. 
+
+snippet:5to6-bus-send-publish
+
+
+## Accessing the CurrentMessageContext
+
+Previously it was possible to access message parameters such `MessageId`, `ReplyToAddress` and the message headers via the `CurrentMessageContext` property on the `IBus` interface. These message properties are now available directly via the message handler context parameter.
+
+snippet: 5to6-messagecontext
+
+
+## Dependency Injection
+
+In previous versions of NServiceBus the `IBus` interface was automatically registered in the IoC container. In Version 6, the new context aware interfaces, for example, the `IEndpointInstance`, `IMessageSession` and `IMessageHandlerContext`, etc., will not be automatically registered in the IoC container.  
+
+When a custom component received an instance of `IBus` via dependency injection:
+
+If the custom component was sending messages using the injected `IBus` from outside of message handlers, for example, in an MVC Controller class, then it is safe to register the `IEndpointInstance` when self hosting. This interface can then be used to send messages. See [Injecting the Bus into ASP.NET MVC Controller](/samples/web/asp-mvc-injecting-bus/) for an example of this.
+
+If the custom component is accessed from within message handlers then the `IMessageHandlerContext` parameter should be passed to the custom component instead of the `IEndpointInstance` interface to either send or publish messages. 
+
+It is not safe to simply replace all instances of `IBus` with `IEndpointInstance`. If the `IEndpointInstance` interface is used inside a message handler to send or publish messages, then:
+
+- those messages will not participate in the same transaction scope as that of the message handler. This could result in messages dispatched or events published via the `IEndpointInstance` interface even if the message handler resulted in an exception and the operation was rolled back.
+
+- those messages will not be part of the [batching operation](/nservicebus/messaging/batched-dispatch.md).
+
+- those messages will not contain any important message header information that is available via the `IHandlerMessageContext` interface parameter, e.g., CorrelationId.  
```

Comment:
A message is constituted of: message body + message headers i.e. the envelope details. Message headers are part of the message, just like a body of a message is. If there's a confusion, what do you think will be better? It helps if you add your suggestion.